### PR TITLE
Realm Web: Minor refactors

### LINKED ISSUE #2962: Realm Web: Various minor refactors
After the beta launch of Realm Web, we have a few minor refactors that we want to complete:
* [x] Align types between Realm JS and Realm Web
* [x] Expose functions on the `User`.
* [x] Move the `serialize` method from services into a transport to make the services more reusable between Realm JS and Realm Web.
* [x] Remove the `Remote` prefix from the MongoDB Client classes.
* [ ] ~Add the the missing credentials types to app.d.ts.~
* [x] Revisit and potentially align Remote MongoDB API with the Node.js driver.
* [ ] Rename the existing `logOut` to `logOutConfirmed` and delete the access and refresh tokens only after session deletion has succeeded.
* [ ] Create a new synchronous `logOut` which deletes access and refresh tokens and optimistically deletes the session (just console.warn-logging any errors).
  * Consider what happens when a different tab invalidates the session ...
* [ ] Make the `Credentials` type argument default to `any`.

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This closes #2962

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

### CODE REVIEW

File: packages/realm-web/src/FunctionsFactory.ts [hunk 1]
```diff
@@ -67,10 +67,52 @@ function cleanArgs(args: any[]) {
...
+}
+
 /**
  * Defines how functions are called
  */
 export class FunctionsFactory {
+    /**
+     * Create a factory of functions, wrapped in a Proxy that returns bound copies of `callFunction` on any property.
+     *
+     * @param transport The underlying transport to use when requesting
```
Comment (kneth) [inline packages/realm-web/src/FunctionsFactory.ts:88]:
```suggestion
     * @param transport The underlying transport to use when requesting.
```

File: packages/realm-web/src/FunctionsFactory.ts [hunk 2]
```diff
@@ -67,10 +67,52 @@ function cleanArgs(args: any[]) {
...
+     * @param config Additional configuration parameters
```
Comment (kneth) [inline packages/realm-web/src/FunctionsFactory.ts:89]:
```suggestion
     * @param config Additional configuration parameters.
```

File: packages/realm-web/src/User.test.ts [hunk 3]
```diff
@@ -90,6 +90,20 @@ describe("User", () => {
...
     });
 
+    it("expose a functions factory", async () => {
+        const user = new User({
+            app: new MockApp("my-mocked-app", [{ pong: "ball" }]),
+            id: "some-user-id",
+            accessToken: "deadbeef",
+            refreshToken: "very-refreshing",
+        });
+        // Refresh the profile and expect a firstName
```
Comment (steffenagger) [inline packages/realm-web/src/User.test.ts:100]: This comment seems a bit odd?

**Comment by kraenhansen** [comment]
I just released this as Realm Web v0.5.0.

**Pull request merged**